File: code/generate_strategies/IR(summary)/python/python/function_algorithms_search_linear_search.py
from algorithms.search.linear_search import linear_search

class LinearSearchFunction:
    def linear_search_from_string(self, arr, target):
        if not arr:
            return -1
        arr = list(map(int, arr.split(',')))
        target = int(target)
        return linear_search(arr, target)

File: code/generate_strategies/RA(method)/java/function_algorithms_search_linear_search.java
package com.example;

import java.util.Arrays;
import org.apache.commons.lang3.StringUtils;

public class LinearSearchFunction {
    public int linear_search_from_string(String arr, String target) {
        if (StringUtils.isEmpty(arr)) {
            return -1;
        }

        int[] numbers = Arrays.stream(arr.split(","))
                .mapToInt(Integer::parseInt)
                .toArray();
        int targetNum = Integer.parseInt(target);

        for (int i = 0; i < numbers.length; i++) {
            if (numbers[i] == targetNum) {
                return i;
            }
        }
        return -1;
    }
}